qemu/tests/mq_change_qnum.py

```diff
@@ -33,26 +33,35 @@ def change_queues_number(session, ifname, q_number, queues_status=None):
         if not queues_status:
             queues_status = get_queues_status(session, ifname)
 
-        if (q_number != queues_status[1] and q_number <= queues_status[0] and
-                q_number > 0):
-            expect_status = 0
-        else:
-            expect_status = 1
-
         status, output = session.cmd_status_output(mq_set_cmd, safe=True)
         cur_queues_status = get_queues_status(session, ifname)
-        if status != expect_status:
-            err_msg = "Change queues number failed, "
-            err_msg += "current queues set is %s, " % queues_status[1]
-            err_msg += "max allow queues set is %s, " % queues_status[0]
-            err_msg += "when run cmd: '%s', " % mq_set_cmd
-            err_msg += "expect exit status is: %s, " % expect_status
-            err_msg += "output: '%s'" % output
-            test.fail(err_msg)
-        if not status and cur_queues_status == queues_status:
-            test.fail("params is right, but change queues failed")
-        elif status and cur_queues_status != queues_status:
-            test.fail("No need change queues number")
+
+        if q_number == queues_status[1]:
+            if cur_queues_status != queues_status:
+                err_msg = "The queues number should not be changed,"
+                err_msg += "as original value is alreay equal to "
+                err_msg += "expected value."
+                test.fail(err_msg)
```

Comment:
I'm wondering whether we should use test.fail here.
Before a specific version(I cannot recall which one), no error returns even if the user want to change queues to a value it's already be.
I'd suggest that we only give log, not to fail the test.

Reply:
@xiywang Thanks for review.  You are right, some version return with error, some not. However, the queues should not be changed when setting a value it's already be, right? If it changes, no matter whether it returns error or not, it must be a  product bug. That's why we should use test.fail here. Do you agree?

Reply:
Say the current queue is 4, and the user want to change it to 4. In this circumstance, no matter the cmd line returns error or not, the operation is acceptable in my opinion. We only talk about the intention of the change.
The only thing that matters is, whether our product returns error or not. It if returns error, we can guess that it has some comparison like: if (exp_val == val) then print error log. And the previous one must be lack of this comparison.
In previous code, the queue did changed, to a value it was already be. But from the result perspective, did the value really changed? No, it does not. It's still 4. Not anything else, not 3 or 2. That 4 is what we expected. So we cannot say it's a product bug because the result is right. But a little off performance because a redundant cmd is executed.
In  conclusion, I do not think we should give test.fail. You can still correct me if you do not agree.

Reply:
Sorry, your code is right, we should fail the test.
But the log should not mention the value already was what we expected, that's misleading. Let's just tell user that the queue changed into wrong value.
The circumstance of this part is: let's assume that the old value is 4, we want to change it into 4 (line 39, check). But the current queue status is [3][4] (for eg) and we are expecting [4][4].
So the log of "The queue numbers are wrong, we are expecting [?][?], but the current value is [?][?]".
Is this more understandable?

Reply:
@xiywang  Sounds much better. I'll improve the log info later. Thanks a lot.